packages/graphback-core/src/relationships/relationshipHelpers.ts

```diff
@@ -45,59 +45,6 @@ export function isOneToManyField(field: GraphQLField<any, any>): boolean {
   return !!oneToManyAnnotation
 }
 
-/**
- * Strips all relationship annotations from a string
- * @param fieldDescription
- */
-export const stripRelationshipAnnotation = (fieldDescription: string = '') => {
-  if (!fieldDescription.includes('\n')) {
-    return '';
-  }
-
-  const strippedDescription = fieldDescription.split('\n').filter((line: string) => !parseRelationshipAnnotation(line));
-
-  return strippedDescription.join('\n');
-}
-
-/**
- * Strips all non-relationship annotations from a string
- *
- * @param fieldDescription
- */
-export const getRelationshipAnnotationString = (fieldDescription: string = '') => {
-  if (!fieldDescription.includes('\n') && !parseRelationshipAnnotation(fieldDescription)) {
-    return '';
-  }
-
-  const filteredDescription = fieldDescription.split('\n').filter(parseRelationshipAnnotation);
-
-  return filteredDescription.join('\n').trim();
-}
-
-/**
- * Helper to merge two description strings which may or may not have a relationship annotation.
- * This helper keeps non-relationship annotations and merges them together.
- * It chooses the relationship annotation with the `key` field when merging.
- *
- * @param generatedDescription
- * @param customDescription
- */
-export const mergeDescriptionWithRelationshipAnnotation = (generatedDescription: string, customDescription: string) => {
-  const descriptionLines = [stripRelationshipAnnotation(generatedDescription), stripRelationshipAnnotation(customDescription)];
-
-  for (const description of [customDescription, generatedDescription]) {
-    const relationshipDescription = getRelationshipAnnotationString(description);
-    const parsedAnnotation = parseRelationshipAnnotation(description);
-
-    if (parsedAnnotation && parsedAnnotation.key) {
-      descriptionLines.push(relationshipDescription);
-      break;
-    }
-  }
-
-  return descriptionLines.join('\n').trim();
-}
```

Comment:
I think this function can be optimized for sure. It is needed to ensure the annotation is merged with the field description (that is not an annotation)

Reply:
Yes, a first attempt of that is:

https://github.com/aerogear/graphback/blob/6f80afa191444414d8d89d14a8198f2c3ae4d12a/packages/graphback-core/src/relationships/RelationshipMetadataBuilder.ts#L151-L196

We make sure that the annotation with the needed arguments i.e (key and field) are there. 

This comes with one problem, the original annotation expression is never stripped out: see https://github.com/aerogear/graphback/pull/1818#discussion_r465598228

Reply:
I can see from the snapshots that this problem still exists. Are there plans to fix it?

Reply:
Ah sorry, I missed this. 

I do not think this a big issue and worth the effort to be honest.

Reply:
Or `graphql-metadata` would refuse this when parsing the `annotation` since it is declared twice?

Reply:
I am not sure what happens if you use `graphql-metadata` to parse the annotation (which one does it choose). We want the one with `key`.

If it handles that correctly, yes I get that it is not a massive issue, however the final schema could look a bit confusing. I will do a verification again.

Reply:
`graphql-metadata` would use the first annotation hence the reason behind it (the generated annotation) coming first before the `oldDescription`. It is more of the output schema. To make it clear, we could a `xxx generated by graphback`. 
 
Also, when the annotation does contain the `key` field, this is  block will be ignored and the original description will be returned.